src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -1015,8 +1015,80 @@ zocl_load_aie_only_pdi(struct drm_zocl_dev *zdev, struct axlf *axlf,
 int
 zocl_pl_only_reset(struct drm_zocl_dev *zdev, const char *buf, size_t count)
 {
-	/* TODO */
-	DRM_ERROR("%s PL Reset support doesn't added yet", __func__);
+	const struct list_head *ptr = NULL;
+	struct drm_zocl_slot *slot = zdev->pr_slot[ZOCL_DEFAULT_XCLBIN_SLOT];
+	struct kds_sched *kds = NULL;
+	struct kds_client *client = NULL;
+	struct kds_client_ctx *curr = NULL;
+	struct kds_client_ctx *tmp = NULL;
+	void __iomem *map = NULL;
+	int ret = 0;
+
+	if (!slot)
+	       return count;
+
+	if (slot->xclbin_type == ZOCL_XCLBIN_TYPE_FULL)
+		return count;
+
+	if (strcmp(zdev->zdev_data_info->fpga_driver_name, "versal_fpga"))
+		return count;
+
+	mutex_lock(&slot->slot_xclbin_lock);
+	kds = &zdev->kds;
+
+	/* Find out number of active client and free all the context
+	 * associated with it
+	 */
+	list_for_each(ptr, &kds->clients) {
+		client = list_entry(ptr, struct kds_client, link);
+		kds_fini_pl_only_client(kds, client);
+
+		/* Delete all the existing context associated to this device for this
+		 * client.
+		 */
+		list_for_each_entry_safe(curr, tmp, &client->ctx_list, link) {
+			if (!zocl_xclbin_same_uuid(slot, curr->xclbin_id))
+				continue;
+
+			/* Unlock this slot specific xclbin */
+			zocl_xclbin_release(slot, curr->xclbin_id);
+			vfree(curr->xclbin_id);
+			list_del(&curr->link);
+			vfree(curr);
+		}
+	}
```

Comment:
Can we put this into a function? If this is common code, please define the function in kds_core.c. If this is zocl specific, please put it in zocl_kds.c.

Reply:
The implementation has been changed. Please re-review once again.